NYPLAudiobookToolkit/Bookmarks/NYPLAudiobookBookmarkFactory.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+import NYPLUtilities
+
+public final class NYPLAudiobookBookmarkFactory {
+  class public func parseLocatorString(
+    _ selectorValueEscJSON: String) -> (title: String?, part: UInt, chapter: UInt, audiobookId: String, duration: Double, time: Double)? {
+      
+      // Convert string to JSON object
+      guard
+        let selectorValueData = selectorValueEscJSON.data(using: String.Encoding.utf8),
+        let selectorValueJSON = (try? JSONSerialization.jsonObject(with: selectorValueData)) as? [String: Any]
+      else {
+        ATLog(.error, "Error serializing locator. SelectorValue=\(selectorValueEscJSON)")
+        return nil
+      }
+
+      // Extract data from JSON
+      guard
+        let part = selectorValueJSON[NYPLBookmarkSpec.Target.Selector.Value.audiobookLocatorPartKey] as? UInt,
+        let chapter = selectorValueJSON[NYPLBookmarkSpec.Target.Selector.Value.audiobookLocatorChapterKey] as? UInt,
+        let audiobookId = selectorValueJSON[NYPLBookmarkSpec.Target.Selector.Value.audiobookLocatorBookIDKey] as? String,
+        let duration = selectorValueJSON[NYPLBookmarkSpec.Target.Selector.Value.audiobookLocatorDurationKey] as? Double,
+        let time = selectorValueJSON[NYPLBookmarkSpec.Target.Selector.Value.audiobookLocatorOffsetKey] as? Double
+      else {
+        ATLog(.error, "Locator does not contain required value. SelectorValue=\(selectorValueEscJSON)")
+        return nil
+      }
+    
+      let title = selectorValueJSON[NYPLBookmarkSpec.Target.Selector.Value.audiobookLocatorTitleKey] as? String
+      
+      return (title: title, part: part, chapter: chapter, audiobookId: audiobookId, duration: duration, time: time)
+  }
+  
+  // Create a locator string to be store as the selector value of a bookmark
+  class public func makeLocatorString(title: String,
```

Comment:
it would be good (and should be easy) to add a couple unit tests for this function.

Reply:
I have added some unit tests at `NYPLBookmarkSpecTests` in Simplified-iOS repo, should I move it over?

Reply:
if they test this function, yeah, that would be good.